Mathlib/Data/Polynomial/Coeff.lean

```diff
@@ -300,10 +300,26 @@ theorem mul_X_pow_injective (n : ℕ) : Function.Injective fun P : R[X] => X ^ n
   rw [← coeff_X_pow_mul P n i, hPQ, coeff_X_pow_mul Q n i]
 #align polynomial.mul_X_pow_injective Polynomial.mul_X_pow_injective
 
+theorem pow_X_mul_injective (n : ℕ) : Function.Injective fun P : R[X] => P * X ^ n := by
```

Comment:
This would be nice to state as `RightRegular (X^n)` (which is defeq) or similar

Reply:
I think that `Polynomial.Monic.isRegular` exists as well.

Reply:
Looking at `Polynomial.Monic.isRegular` I think that the proof uses subtractions, whereas the result should be true for semirings.  In fact, assuming that the leading coefficient is regular should be enough.  The proof should go via Polynomial.coeff_mul_degree_add_degree.

Reply:
Just to make my suggestion be concrete
```suggestion
theorem X_pow_isRegular (n : ℕ) : IsRegular (X ^ n) := by
```
from which this result and the symmetric one is a single projection away (`.left` and ` .right`)

Reply:
Btw, my comment about was wrong: I confused `IsRegular` with not being a zero-divisor.  To atone, I PRed a counterexample: #6476